alibi_detect/saving/saving.py

```diff
@@ -132,33 +134,39 @@ def _save_detector_config(detector: Detector, filepath: Union[str, os.PathLike])
         filepath.mkdir(parents=True, exist_ok=True)
 
     # Get the detector config (with artefacts still within it)
-    cfg = getattr(detector, 'get_config', None)
-    if cfg is not None:
-        cfg = cfg()  # TODO - can just do detector.get_config() once all detectors have a .get_config()
+    if hasattr(detector, 'get_config'):
+        cfg = detector.get_config()  # type: ignore[union-attr]  # TODO - remove once all detectors have get_config
```

Comment:
We should accumulate all these todo's in an issue somewhere once this is merged so that when stuff gets fixed we know what to update